Appraisals

```diff
@@ -2,36 +2,40 @@ lib = File.expand_path('../lib', __FILE__)
 $LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
 require 'ddtrace/version'
 
+def ruby_version?(version)
+  full_version = "#{version}.0" # Turn 2.1 into 2.1.0 otherwise #bump below doesn't work as expected
+
+  Gem::Version.new(full_version) <= Gem::Version.new(RUBY_VERSION) &&
+    Gem::Version.new(RUBY_VERSION) < Gem::Version.new(full_version).bump
+end
+
 def self.gem_cucumber(version)
   appraise "cucumber#{version}" do
     gem 'cucumber', ">=#{version}.0.0", "<#{version + 1}.0.0"
   end
 end
 
-if Gem::Version.new(RUBY_VERSION) < Gem::Version.new(Datadog::VERSION::MINIMUM_RUBY_VERSION)
-  raise NotImplementedError, "Ruby versions < #{Datadog::VERSION::MINIMUM_RUBY_VERSION} are not supported!"
-elsif Gem::Version.new('2.1.0') <= Gem::Version.new(RUBY_VERSION) \
-      && Gem::Version.new(RUBY_VERSION) < Gem::Version.new('2.2.0')
+if ruby_version?('2.1')
```

Comment:
Do you think `case/when` would be a step too far here?

Reply:
Since it's a bit "too long" I like that `if`/`elsif` is a bit more explicit, but I'm not against it either.

It also came to mind that we could just use some DSL-like blocks, something like

```ruby
on_ruby('2.1') do
   ...
end
```

but I didn't go forward with it (yet?). I may do some crazier cleanups one day ;P